src/Process.php

```diff
@@ -120,11 +122,15 @@ public function start(LoopInterface $loop, $interval = 0.1)
             });
         };
 
-        $this->stdin  = new Stream($this->pipes[0], $loop);
-        $this->stdin->pause();
-        $this->stdout = new Stream($this->pipes[1], $loop);
+        $this->stdin  = new WritableResourceStream($this->pipes[0], $loop);
+        $this->stdin->on('close', function () use ($that) {
+            if (is_resource($that->pipes[0])) {
+                fclose($that->pipes[0]);
+            }
+        });
```

Comment:
I suppose this is work around for https://github.com/reactphp/stream/pull/107? Does it make sense to back port this to older stream release and only target fixed versions here?